src/unittests/floating_point.spec.ts

```diff
@@ -2079,15 +2079,17 @@ const kULPErrorValue = {
 g.test('ulpInterval')
   .params(u =>
     u
-      .combine('trait', ['f32', 'f16'] as const)
+      .combine('trait', ['abstract', 'f32', 'f16'] as const)
       .beginSubcases()
       .expandWithParams<ULPCase>(p => {
-        const constants = FP[p.trait].constants();
-        const ULPValue = kULPErrorValue[p.trait];
-        const plusOneULP = kPlusOneULPFunctions[p.trait];
-        const plusNULP = kPlusNULPFunctions[p.trait];
-        const minusOneULP = kMinusOneULPFunctions[p.trait];
-        const minusNULP = kMinusNULPFunctions[p.trait];
+        // For ULP purposes, abstract float behaves like f32, so swizzling it in.
+        const trait = p.trait === 'abstract' ? 'f32' : p.trait;
```

Comment:
How about introducing some kind of FP mapping dictionary for ULP propose, instead of using `?:` in multiple places?

Something in the file scope like
``` typescript
// For ULP purposes, abstract float behaves like f32, so swizzling it in.
const kFPTraitForULP = {
  abstract: 'f32',
  f32: 'f32',
  f16: 'f16',
} as const;
```
and then use e.g.
``` typescript
        const constants = FP[kFPTraitForULP[p.trait]].constants();
        const ULPValue = kULPErrorValue[kFPTraitForULP[p.trait]];
        const plusOneULP = kPlusOneULPFunctions[kFPTraitForULP[p.trait]];
        const plusNULP = kPlusNULPFunctions[kFPTraitForULP[p.trait]];
        const minusOneULP = kMinusOneULPFunctions[kFPTraitForULP[p.trait]];
        const minusNULP = kMinusNULPFunctions[kFPTraitForULP[p.trait]];
```
or
``` typescript
        const trait = kFPTraitForULP[p.trait];
        const constants = FP[trait].constants();
        const ULPValue = kULPErrorValue[trait];
        const plusOneULP = kPlusOneULPFunctions[trait];
        const plusNULP = kPlusNULPFunctions[trait];
        const minusOneULP = kMinusOneULPFunctions[trait];
        const minusNULP = kMinusNULPFunctions[trait];
```

Reply:
Done